src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,325 @@
+<?php
+namespace App\Api\Test;
+
+use ApiPlatform\Core\Bridge\Symfony\Bundle\Test\Client as TestClient;
+use App\Api\Test\BaseApiTestCase;
+use App\Entity\Client;
+use App\Entity\Script;
+
+class ClientTest extends BaseApiTestCase
+{
+    public function testCreateClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::named($clientName);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($clientJson);
+        $this->assertHydraContext();
+    }
+
+    public function testCreateClientAllParameters(): void 
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $scriptId = $this->getScriptId($httpClient, 'script_all_true');
+        $clientJson = ClientMother::withAllParameters(
+            $clientName, 
+            1, 
+            "some description", 
+            false, 
+            2, 
+            [$scriptId], 
+            [$scriptId],
```

Comment:
Mantra: "Test to find the error not the correctness"

Saving the prescript as a postscript and viceverse would be quite an easy mistake to do, so don't be lazy and use different scirpts.

Sending just a single script isn't the best approach either, it wouldn't be hard to make a mistake that treats the first element in the array well but misses or overlaps the other.

Please don't be so self-satisifed nor lazy.